<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "http://www.eclipse.org/jetty/configure.dtd">

<Configure id="fcrepo" class="org.eclipse.jetty.webapp.WebAppContext">
  <Set name="contextPath">/fcrepo</Set>
  
  <!-- Use line below once workaround is removed -->
  <Set name="war"><SystemProperty name="jetty.base" default="."/>/webapps/fcrepo.war</Set>
  <!-- <Set name="war"><SystemProperty name="jetty.base" default="."/>/webapps/fcrepo</Set> -->
  
  <!-- Remove line below when workaround is removed-->
  <!-- <Set name="overrideDescriptor"><SystemProperty name="jetty.base" default="."/>/overlay-web.xml</Set> -->

  <!--
    Added because we are hitting this issue w/ v5:
    https://stackoverflow.com/questions/41995029/jersey-spring3-2-25-1-produces-failed-startup-of-context-error-in-jetty-9-3
  -->
  <Call name="setAttribute">
    <Arg>org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern</Arg>
    <Arg>.*/^(asm-all-repackaged)[^/]*\.jar$</Arg>
  </Call>

  <Get name="securityHandler">
    <Set name="authenticator">
      <New class="edu.ucdavis.library.jetty.JwtAuthenticator"></New>
    </Set>
  </Get>
</Configure>